xonsh/tools.py

```diff
@@ -1171,7 +1171,7 @@ def backup_file(fname):
     import shutil
     from datetime import datetime
     base, ext = os.path.splitext(fname)
-    newfname = base + '.' + datetime.now().isoformat() + ext
+    newfname = '%s.%s%s' % (base, datetime.now().isoformat().replace(':', '-'), ext)
```

Comment:
Also, it looks like this change kicked this line over the PEP8 limit of 80 characters per line.  Maybe we could assign that timestamp to a variable on the line before, and then reference that here?